api/v1alpha3/common_types.go

```diff
@@ -36,6 +37,18 @@ const (
 	// Controllers working with Cluster API objects must check the existence of this annotation
 	// on the reconciled object.
 	PausedAnnotation = "cluster.x-k8s.io/paused"
+
+	// BootstrapSecretType defines the type of secret the kubeadm bootstrap
+	// controller will create.
+	BootstrapSecretType corev1.SecretType = "bootstrap.cluster.x-k8s.io/data" // #nosec
+
+	// ControlPlaneSecretType defines the type of secret the kubeadm
+	// control-plane controller will create.
+	ControlPlaneSecretType corev1.SecretType = "control-plane.cluster.x-k8s.io/data" // #nosec
+
+	// ClusterSecretType defines the type of secret the cluster controller
+	// will create.
+	ClusterSecretType corev1.SecretType = "cluster.x-k8s.io/data" // #nosec
```

Comment:
We can likely consolidate all of these on `ClusterSecretType`, since they are all part of the "Cluster API contract" with providers/users/etc.

Reply:
Can you please add a comment for why `#nosec` is needed here :)

Reply:
+1

Reply:
If we're doing one secret for all, I'd suggest to change the type string to `cluster.x-k8s.io/secret`

Reply:
I guess we could just use one secret type for all the secrets created by CAPI which would make things a whole lot simpler but I thought the point of the `SecretType` was to not just provide ownership distinction of the secrets but also differentiate how a secret should be handled (if necessary). 
As per the godoc,
> Used to facilitate programmatic handling of secret data.

I'm fine with it either way, just wanted to make sure.

Reply:
I think if we get too granular with it we create a few different issues for ourselves:
- User confusion regarding which type to create which secret when creating manually
- Potential disparity in which type to set if using KubeadmControlPlane vs not
- Conflicts on which type to set if a Secret is needed for more than one use (for example the kubeconfig secret).

We can always move towards a more granular approach later if we see the need as well.